[health-check] fix performance issue and add extra enhacements

meaksh opened the pull request:
## What does this PR change?

This PR fixes the performance issue found on supportconfigs due unexpected tons of Salt jobs. Additionally it does a couple of other fixes. See each commit individually.

NOTE: This PR is targetting `health-check-skeleton` feature branch.

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

m-czernek commented on python/health-check/src/health_check/containers/promtail/Dockerfile:
```diff
@@ -1,8 +1,7 @@
 FROM opensuse/leap:latest
-#FROM registry.suse.com/bci/python:3.11
```
@ycedres did we settle in on the actual base images in the end?

m-czernek replied:
(I think Yeray is on PTO, we can probably return to this later)

m-czernek commented on python/health-check/src/health_check/main.py:
```diff
@@ -94,14 +109,14 @@ def run(ctx: click.Context, from_datetime: str, to_datetime: str, since: int):
 
 def main():
-    console.print(Markdown("# Health Check"))
+    console.print(Markdown("# Uyuni Health Check"))
```
Same as above

m-czernek commented on python/health-check/src/health_check/config/templates/grafana_dashboard/supportconfig_with_logs.template.json:
```diff
@@ -70,9 +83,9 @@
+      "id": 71,
       "panels": [],
-      "title": "Memory",
+      "title": "Uyuni Server and Salt",
```
Note, same as above - I wanted to remove all `uyuni` occurrences from the Grafana board as well.

m-czernek commented on python/health-check/src/health_check/config/templates/grafana_dashboard/supportconfig_with_logs.template.json:
```diff
@@ -1380,13 +1343,13 @@
   "time": {
-    "from": "",
-    "to": ""
+    "from": "now-30d",
+    "to": "now"
```
Is this intentional change? We leave this empty because health-check tool sets this based on the passed-in parameters

m-czernek reviewed (commented):
Additionally, can you add just a brief description of what you're modifying in the Grafana board? Not sure if you're just moving/renaming stuff, or if you're removing/modifying the view.

meaksh commented on python/health-check/pyproject.toml:
```diff
@@ -29,7 +29,7 @@ homepage = "https://github.com/uyuni-project/uyuni"
 
 [project.scripts]
-health-check = "health_check.main:main"
+uyuni-health-check = "health_check.main:main"
```
Thanks for clarifying this was actually intentional (I thought the rename was only related to the codebase files, but not for the CLI).

It just feels a bit weird to me that it doesn't have anything on the CLI name that refers to the what "health"  this is checking :) 

Should be maybe use `mgr-health-check` for the CLI tool?

m-czernek replied:
Well, in my mind, "what" health are we checking is covered by the fact that you install/deploy it from Uyuni or MLM repositories. So if you're deploying it, you know exactly what health are you checking.

That said, I really like `mgr-health-check` so that we group all MLM/Uyuni scripts under one "namespace".

meaksh commented:
@m-czernek I just addressed your comments from the review.

About the changes on the dashboard:
- Rename sections/rows titles.
- Refactor Salt keys panel.
- Add Java configuration metrics.
- Added "misc" and "hw" metrics.
- Adjust panels to fit the entire window width.

Here are some screenshots:

![Captura desde 2025-03-03 10-11-30](https://github.com/user-attachments/assets/2ba1c47f-6c71-4b65-b843-3a807de4b1ec)
![Captura desde 2025-03-03 10-11-43](https://github.com/user-attachments/assets/6c3e8f45-3bf9-42b4-a27e-3ce39f1d6197)

meaksh commented:
@m-czernek I just addressed your comments from the PR review:

About the changes made on the dashboards:
- Refactor Salt Keys panels
- Add Java configuration metrics
- Add "misc" and "hw" metrics
- Adjust panels to fit entire windows width
- Rename sections/rows titles

Here are some screeshots:

![Captura desde 2025-03-03 10-11-30](https://github.com/user-attachments/assets/f764694c-178e-4698-b3de-e0bf69e39537)
![Captura desde 2025-03-03 10-11-43](https://github.com/user-attachments/assets/65bfddf3-435b-48c1-acb4-ac0d25477c3d)

m-czernek commented on python/health-check/src/health_check/main.py:
```diff
@@ -60,7 +62,7 @@ def cli(ctx: click.Context, supportconfig_path: str, verbose: bool):
 )
 @click.pass_context
-def run(ctx: click.Context, from_datetime: str, to_datetime: str, since: int):
+def start(ctx: click.Context, from_datetime: str, to_datetime: str, since: int):
```
Does this imply change in the docs? I.e. right now, we say to run `health-check run ...` - do we have to change it to `health-check start ...`, or is it just `health-check ...`? Either way, we'll probably need to modify the readme, right?

meaksh replied:
That's true, we need to change the README and wiki documentation.

I'll include the documentation changes and some of your suggestions in a follow-up PR.

m-czernek reviewed (approved):
This LGTM now - adding +1, but please also check whether readme doesn't need a change, since we're changing the `run/clean` method names.

Additionally, I'm not sure we want to tackle this within this PR, but I have a few suggestions regarding the dashboard.

WDYT about:

- Removing:
  -  worker threads, socket pool size, timeout, and gather job timeout
  - Java settings

Reason for this is that I'm not sure this is very helpful to viewers. What is helpful is the rules we have based on those values.

- Then, we might want to break out `num_of_channels` from miscellaneous as a separate counter (IMO this is interesting/important info to see, similar to num of CPUs and RAM)
- I wonder if we might make some transformation on `master/proxy/client`, where `1` is essentially `true` and `0` is `false`. Note that we cannot modify these actual values since we use the `1` and `0` in math expressions in some rules (e.g. checking minimal requirements). 

  But, we could either do a frontend transformation on the table, if possible, or do something else, e.g. create a new metric for front-end users. This way, it looks like the supportconfig has 1 server, 0 proxies and 0 clients, which might be a bit confusing.

meaksh commented:
> WDYT about:
> 
> * Removing:
>   
>   * worker threads, socket pool size, timeout, and gather job timeout
>   * Java settings
> 
> Reason for this is that I'm not sure this is very helpful to viewers. [...]

I think we should not remove worker threads, socket pool size, timeout, and gather job timeout metrics, neither the Java configuration values. IMO it is worth to have this complete and agreggated view of the Java configurations values in the dashboard.

Of course, if we have alerts already implemented to indicate some possible known issue, that is great, and we should have as many as possible, but having the actual values displayed here I think helps when debugging to identify not known issues, where we don't have alerts already implemented.

m-czernek commented:
I think the problem I have is that with some metrics, you can't even guess what the proper fully-qualified name is. With some metrics, like worker threads, this is not an issue. But with something like "timeout", this is useless - what kind of timeout? 

I added the Salt config in the beginning when I didn't have a clear idea of what I'm doing. The Java configs are similarly confusing.

meaksh commented:
> I think the problem I have is that with some metrics, you can't even guess what the proper fully-qualified name is. With some metrics, like worker threads, this is not an issue. But with something like "timeout", this is useless - what kind of timeout?
> 

I understand your point here, but I think we could just change the panels so it is clear that these are Salt Master configuration parameters and values. To me, it is still important that you see these values in the dashboard.

> I added the Salt config in the beginning when I didn't have a clear idea of what I'm doing. The Java configs are similarly confusing.

Hmm, ok, now I realized that the names of the Java setting displayed here are not actually the real name of the Java property. Of course, this is then confusing as you mentioned. I would suggest then that we adjust the metrics names provided by the exporter to match the real Java parameter name.
